[CI] GCC Annotations not present for PR #6133

There was some violation of const-ness in this PR, which was a GCC warning but not an error.

Yet for some reason the annotation of the warning is missing from the PR.

Reviewing the `GCC Warnings & Errors` GH Workflow, and the `build_oai` sub-job's logs (available as an `Artifact` in the [job page](https://github.com/magma/magma/pull/6133/checks?check_run_id=2349682691) - I do see the warnings emitted to log:

```
/magma/lte/gateway/c/oai/tasks/nas/esm/esm_information.c:136:42: warning: passing argument 1 of 'clear_protocol_configuration_options' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  136 |     clear_protocol_configuration_options(pco);
      |                                          ^~~
```

But this log is purposefully not emitted to stdout in this GH Action job - because GitHub annotations have severe limitations on count of annotations. So I manually filter the emissions by modified files using git diff.

Log tell from the job indicates that the changed file filtering is broken for this PR.

```
No pattern or formatter specified
...
No pattern or formatter specified
```

This error message comes from the Golang [xo tool](https://github.com/ezekg/xo/blob/master/main.go#L33) I'm using to reformat magma compiler paths in warnings (unfortunately, due to build configuration, they contain a /magma/ prefix).

I'm going to attempt manual reproduction.  One concern is that the Git Diff semantics are not ideal in Github - there may be incorrect file changed lists going on here...  Alternatively there could be some issue with the use of `xo`.

I would be best off migrating this filtering to ReviewDog, which I have since discovered and supports much better file filtering.

--- a/.github/workflows/gcc-problems.yml
+++ b/.github/workflows/gcc-problems.yml
@@ -115,7 +115,7 @@ jobs:
             CPPFLAGS="-Wextra -Wshadow -Wimplicit-fallthrough -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wnull-dereference -Wjump-misses-init -Wformat=2 -Wstrict-overflow=4 -Wuninitialized -Wshift-overflow=2" make build_oai 2>&1 > /magma/compile.log
             BUILD_EXIT_CODE=$?
             for file in ${{ steps.changed_files.outputs.all }};
-            do grep "$file" /magma/compile.log | xo "/\/magma\/((.*):(\d+):(\d+):\s+(?:fatal\s)?(warning|error):\s+(.*))/$1/" || true;
+            do grep "$file" /magma/compile.log | xo '/\/magma\/((.*):(\d+):(\d+):\s+(?:fatal\s)?(warning|error):\s+(.*))/$1/' || true;
             done;
             exit $BUILD_EXIT_CODE
       -
@@ -175,7 +175,7 @@ jobs:
             CPPFLAGS="-Wextra -Wshadow -Wimplicit-fallthrough -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wnull-dereference -Wjump-misses-init -Wformat=2 -Wstrict-overflow=4 -Wuninitialized -Wshift-overflow=2" make build_session_manager 2>&1 > /magma/compile.log
             BUILD_EXIT_CODE=$?
             for file in ${{ steps.changed_files.outputs.all }};
-            do grep "$file" /magma/compile.log | xo "/\/magma\/((.*):(\d+):(\d+):\s+(?:fatal\s)?(warning|error):\s+(.*))/$1/" || true;
+            do grep "$file" /magma/compile.log | xo '/\/magma\/((.*):(\d+):(\d+):\s+(?:fatal\s)?(warning|error):\s+(.*))/$1/' || true;
             done;
             exit $BUILD_EXIT_CODE
       -

